apps/language_server/lib/language_server/providers/completion.ex

```diff
@@ -593,6 +601,51 @@ defmodule ElixirLS.LanguageServer.Providers.Completion do
     end
   end
 
+  def suggest_module_name(file_path) when is_binary(file_path) do
+    file_path
+    |> Path.split()
+    |> Enum.reverse()
+    |> do_suggest_module_name()
+  end
+
+  defp do_suggest_module_name([]), do: nil
+
+  defp do_suggest_module_name([filename | reversed_path]) do
+    filename
+    |> String.split(".")
+    |> case do
+      [file, "ex"] ->
+        do_suggest_module_name(reversed_path, [file], topmost_parent: "lib")
+
+      [file, "exs"] ->
+        if String.ends_with?(file, "_test") do
+          do_suggest_module_name(reversed_path, [file], topmost_parent: "test")
```

Comment:
@sharpfun Thanks for the suggestions, but I'm hesitant to go any further at this point before anyone of the maintainers chimes in because it might be the wrong way to go about it

Reply:
We can't make this work for everyone. I'd aim for covering basic mix project and phoenix app as those are most commonly used.